src/SFML/Graphics/SpriteBatch.cpp

```diff
@@ -0,0 +1,267 @@
+
+        const auto& vertices = m_batchMode == BatchMode::Deferred ? m_unsortedVertices : m_vertices;
+
+        target.draw(&vertices[startTriangle], m_batches[i].vertexCount, PrimitiveType::Triangles, statesCopy);
+
+        startTriangle += m_batches[i].vertexCount;
+    }
+}
+
+
+////////////////////////////////////////////////////////////
+void SpriteBatch::pushTriangle(const Vertex&    a,
+                               const Vertex&    b,
+                               const Vertex&    c,
+                               const Transform& transform,
+                               const Texture*   texture,
+                               float            depth)
+{
+    m_triangles.emplace_back(texture, depth);
+
+    m_unsortedVertices.emplace_back(transform * a.position, a.color, a.texCoords);
+    m_unsortedVertices.emplace_back(transform * b.position, b.color, b.texCoords);
+    m_unsortedVertices.emplace_back(transform * c.position, c.color, c.texCoords);
+
+    m_updateRequired = true;
+}
+
+
+////////////////////////////////////////////////////////////
+void SpriteBatch::updateBatch() const
+{
+    // Sanity check
+    if (m_triangles.size() == 0)
+        return;
+
+    if (m_batchMode == BatchMode::Deferred)
+    {
+        // Batch based on texture change
+        std::size_t    startIndex  = 0;
+        std::size_t    nextIndex   = 0;
+        const Texture* lastTexture = nullptr;
+
+        while (nextIndex < m_triangles.size())
+        {
+            const Texture* nextTexture = m_triangles[nextIndex].texture;
+            if (nextTexture != lastTexture)
+            {
+                m_batches.emplace_back(lastTexture, (nextIndex - startIndex) * 3);
+                lastTexture = nextTexture;
+                startIndex  = nextIndex;
+            }
+
+            nextIndex++;
+        }
+
+        // Deal with leftovers
+        if (startIndex != m_triangles.size())
+        {
+            m_batches.emplace_back(lastTexture, (m_triangles.size() - startIndex) * 3);
+        }
+    }
+
+    else if (m_batchMode == BatchMode::TextureSort || m_batchMode == BatchMode::DepthSort)
+    {
+        std::vector<std::size_t> indices(m_triangles.size());
+
+        for (std::size_t i = 0; i < m_triangles.size(); i++)
+            indices[i] = i;
+
+        if (m_batchMode == BatchMode::TextureSort)
+        {
+            const auto comp = [&](const std::size_t a, const std::size_t b)
+            { return m_triangles[a].texture < m_triangles[b].texture; };
+
+            std::stable_sort(indices.begin(), indices.end(), comp);
+        }
+
+        else
+        {
+            const auto comp = [&](const std::size_t a, const std::size_t b)
+            {
+                if (m_triangles[a].depth != m_triangles[b].depth)
+                    return m_triangles[a].depth > m_triangles[b].depth;
+                else
+                    return m_triangles[a].texture < m_triangles[b].texture;
```

Comment:
Omitting the braces in this `if/else` helps readability. I'd recommend removing the braces for other simple loops and conditions you've added which only contain a single expression. It halves the total line count and keeps relevant code near each other. Because we have clang-format to check formatting, there is no risk of accidentally writing something like 
```cpp
if (condition)
    run1();
    run2(); // Uh oh, this always get run no matter the value of condition
```

Reply:
I agree in the case of simple loops / `if`s, although in some cases I feel like the extra space given by adding braces helps readability.